components/b2b/AccountHierarchy/AccountHierarchyForm/AccountHierarchyForm.spec.tsx

```diff
@@ -0,0 +1,89 @@
+    setup()
+
+    const parentAccountField: HTMLInputElement = screen.getByRole('textbox', { name: '' })
+    const companyNameField = screen.getByRole('textbox', { name: 'company-name' })
+    const taxIdField = screen.getByRole('textbox', { name: 'tax-id (optional)' })
+    const emaiAddressField = screen.getByRole('textbox', { name: 'email' })
+    const firstNameField = screen.getByRole('textbox', { name: 'first-name' })
+    const lastNameField = screen.getByRole('textbox', { name: 'last-name-or-sur-name' })
+    const submitButton = await screen.findByTestId('submit-button')
+    const cancelButton = await screen.findByTestId('cancel-button')
+
+    expect(parentAccountField).toBeInTheDocument()
+    await waitFor(() =>
+      expect(parseInt(parentAccountField.value)).toBe(Common?.args?.accounts?.[0]?.id)
+    )
+    expect(companyNameField).toBeVisible()
+    expect(taxIdField).toBeVisible()
+    expect(emaiAddressField).toBeVisible()
+    expect(firstNameField).toBeVisible()
+    expect(lastNameField).toBeVisible()
+    expect(submitButton).toBeVisible()
+    expect(cancelButton).toBeVisible()
+  })
+
+  it('should show values entered by user', async () => {
+    const user = userEvent.setup()
+
+    render(<Common {...Common.args} onSave={onSave} onClose={onClose} />)
+
+    const companyNameField: HTMLInputElement = screen.getByLabelText('company-name')
+    const taxIdField: HTMLInputElement = screen.getByLabelText('tax-id (optional)')
+    const emaiAddressField: HTMLInputElement = screen.getByLabelText('email')
+    const firstNameField: HTMLInputElement = screen.getByLabelText('first-name')
+    const lastNameField: HTMLInputElement = screen.getByLabelText('last-name-or-sur-name')
+    const submitButton = await screen.findByTestId('submit-button')
+
+    user.type(companyNameField, 'ABC Enterprise')
+    await waitFor(() => expect(companyNameField.value).toBe('ABC Enterprise'))
+    user.type(taxIdField, '123234')
+    await waitFor(() => expect(taxIdField.value).toBe('123234'))
+    user.type(emaiAddressField, 'aman.shukla@gmail.com')
+    await waitFor(() => expect(emaiAddressField.value).toBe('aman.shukla@gmail.com'))
+    user.type(firstNameField, 'Aman')
+    await waitFor(() => expect(firstNameField.value).toBe('Aman'))
+    user.type(lastNameField, 'Shukla')
+    await waitFor(() => expect(lastNameField.value).toBe('Shukla'))
+
+    user.click(submitButton)
+    await waitFor(() => expect(onSave).toHaveBeenCalledTimes(1))
+  })
+
+  it('should show normal textbox when account added to child account', async () => {
+    render(<Common {...AddAccountToChild.args} onSave={onSave} onClose={onClose} />)
+
+    const companyNameField: HTMLInputElement = screen.getByRole('textbox', {
+      name: 'parent-account',
+    })
+
+    await waitFor(() =>
+      expect(companyNameField.value).toBe(
+        AddAccountToChild?.args?.accounts?.[0]?.companyOrOrganization
+      )
+    )
+  })
+})
```

Comment:
missing stories
should call onSave callback function when user clicks on 'Create Account' button
should call onClose callback function when user clicks on 'Cancel' button

Reply:
Added missing test cases